src/coreclr/jit/lower.cpp

```diff
@@ -4458,6 +4463,16 @@ bool Lowering::TryLowerConditionToFlagsNode(GenTree* parent, GenTree* condition,
         {
             *cond = GenCondition(GenCondition::P);
         }
+
+        if (!allowMultipleFlagsChecks && cond->IsFloat())
+        {
+            const GenConditionDesc& desc = GenConditionDesc::Get(*cond);
+
+            if (desc.oper != GT_NONE)
+            {
+                return false;
+            }
+        }
```

Comment:
On second thought can we just add the support for this? It seems like it just requires emitting an extra `ccmp` for these cases, similar to the extra cmovs sometimes emitted by `genCodeForSelect`. It will be cleaner than having to move `GenConditionDesc` around and adding architecture-specific limitations.